caps/types.go

```diff
@@ -31,6 +31,9 @@ type Type interface {
 	Name() string
 	// Sanitize a capability (altering if necessary).
 	Sanitize(c *Capability) error
+	// Obtain the security snippet for the given security system.
+	// If no security snippet is needed, hand out empty string.
+	SecuritySnippet(c *Capability, securitySystem string) (string, error)
```

Comment:
Considering the data nature, I think the most appropriate result type here is actually a `[]byte`.